src/geometry/partitions/ptscotchunstructuredpartition.h

```diff
@@ -65,47 +76,37 @@ class PTScotchUnstructuredPartition : public Partition<DIM>
     void initIndices(std::vector<SCOTCH_Num>& indices)
     {
         // create 2D grid
-        SCOTCH_Arch arch;
-        SCOTCH_archInit(&arch);
-        SCOTCH_Num *velotabArch;
-        SCOTCH_Num const vertnbrArch = weights.size();
-        velotabArch = new SCOTCH_Num[weights.size()];
-        for (int i = 0; i < vertnbrArch; ++i)
-        {
-            velotabArch[i] = weights[i];
-        }
-        int error = SCOTCH_archCmpltw(&arch, vertnbrArch, velotabArch);
-        if (error) std::cout << "SCOTCH_archCmpltw error: " << error << std::endl;
-
         SCOTCH_Graph graph;
-        error = SCOTCH_graphInit(&graph);
+        int error = SCOTCH_graphInit(&graph);
 
         SCOTCH_Num numEdges = 0;
 
-        for (auto& p : adjacency)
-        {
-            numEdges += p.second.size();
+        // ok, this is SUPER ugly
+        for (auto& p : this->adjacency.getRegion()) {
+            numEdges ++;
```

Comment:
This is not the function you're looking for.

numEdges += this->adjacency.getRegion().size(); ?